### PR TITLE
Gql mock server

### PR DESCRIPTION
**Pull request by daniel-heppner-ibigroup**
**Description:**
Adds a GraphQL Mock server to the Percy tests. 

**PR Checklist:**
- [ ] Does the code follow accessibility standards (WCAG 2.1 AA Compliant)?
- [ ] Are all languages supported (Internationalization/Localization)?
- [ ] Are appropriate Typescript types implemented?

| Before | After |
    |--------|-------|
    |    
image
    |   
image
    |

### CODE REVIEW

**Review by binh-dam-ibigroup — APPROVED** [review]
Looks like good stuff! There are typos here and there, but no major complaints. It would be really nice to print out somewhere the number of calls to of each graphql mock for tracking redundant calls.

**Comment by binh-dam-ibigroup** [comment]
@daniel-heppner-ibigroup Also can you address the translation checks, please?

File: percy/README.md
Comment (daniel-heppner-ibigroup) [reply percy/README.md:44]: This works for me. I think because this is a Bash command, it's okay. I am not sure about Windows.

File: percy/graphql-mocks.js
Comment (daniel-heppner-ibigroup) [reply percy/graphql-mocks.js:45]: I could write something to do that, it isn't a bad idea. I could put some assertions in the code for the number of times we expect each resolver to be called.

File: package.json
Comment (daniel-heppner-ibigroup) [reply package.json:22]: I can add combined to it, but this script runs the express server that hosts GraphQL and REST endpoints.

File: percy/README.md
Comment (daniel-heppner-ibigroup) [reply percy/README.md:52]: Might just remove the mention of IDEs entirely. There are probably multiple ways of doing this.

File: percy/graphql-mocks.js
Comment (daniel-heppner-ibigroup) [reply percy/graphql-mocks.js:45]: Actually I can't do this very easily, the assertions won't work in the mocking server since it's run as a subprocess under the Percy tests. For some reason importing the server and running it directly in the Percy test doesn't work very well, it fails part way through. 
I tried adding some code that logs the calls to each resolver at the end, but the function doesn't seem to be called. 

Open to ideas here.

**Comment by daniel-heppner-ibigroup** [comment]
I'd like to address the translation problems with Nearby View in the parent PR after this is merged into it. This is just to get the Percy Tests all working.

Open to ideas on how to log or document the GraphQL requests made.

**Comment by miles-grant-ibigroup** [comment]
Checkout screenshot 11/12! Something funky is going on there, not sure if it's related to these changes. Overall the result looks quite promising well done. I'm not fussed with graphql logging for now, I think we can deal with it later. Happy to approve this once we confirm what's going on with that itinerary body screenshot

File: percy/har-mock-config.yml
Comment (miles-grant-ibigroup) [inline percy/har-mock-config.yml:26]: Remove overlays section, not needed

File: percy/har-mock-config.yml
Comment (miles-grant-ibigroup) [inline percy/har-mock-config.yml:208]: We don't need this in the har mock config

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
Tiny bit of code cleanup too

**Comment by daniel-heppner-ibigroup** [comment]
After looking into it, I think the missing headsigns in the screenshot is actually correct, as we don't request it in the GraphQL query. The old HAR server would return the same response no matter what the GraphQL query was. At some point I guess we stopped requesting the trip.tripHeadsign in the plan query, but the HAR server kept returning it since we didn't update the response data. The new GraphQL server is smart, so it doesn't return it, which made the test look broken even though it's now actually more correct: we caught a bug!
cc: @binh-dam-ibigroup

**Comment by binh-dam-ibigroup** [comment]
> The new GraphQL server is smart, so it doesn't return it, which made the test look broken even though it's now actually more correct: we caught a bug! cc: @binh-dam-ibigroup

Good finding!

File: percy/har-mock-config.yml
Comment (miles-grant-ibigroup) [inline percy/har-mock-config.yml:102]: This should go

**Review by miles-grant-ibigroup — APPROVED** [review]
One last config cleanup and then I think we are ready to go!

**Pull request merged**
